### DISCUSSION TITLE
Why would adding react-markdown to an iTwinUI project break the project

### DISCUSSION THREAD

**Discussion by candcconsulting** [iTwin/iTwinUI#1893] [Q&A]
I added react-markdown to my project and iTwinUI reports the following errors
ERROR in ./src/App.tsx 11:0-41
Module not found: Error: Can't resolve '@itwin/itwinui-react/styles.css' 
ERROR in ./node_modules/@itwin/appui-react/lib/esm/appui-react/messages/MessageRenderer.js 18:20-30
export 'useToaster' (imported as 'useToaster') was not found in '@itwin/itwinui-react' 
ERROR in ./node_modules/@itwin/appui-react/lib/esm/appui-react/messages/MessageRenderer.js 88:20-30
export 'useToaster' (imported as 'useToaster') was not found in '@itwin/itwinui-react'
Are they not compatible
I thought the css was acceptable
`import '@itwin/itwinui-react/styles.css'`
Please advise
(snippet of packages)
`        "@itwin/itwinui-icons-color-react": "^2.1.0",
        "@itwin/itwinui-react": "^2.7.0",
        "@itwin/measure-tools-react": "^0.13.0",
        "@itwin/presentation-common": "^4.0.0",
        "@itwin/presentation-components": "^4.0.0",
        "@itwin/presentation-frontend": "^4.0.0",
        "@itwin/property-grid-react": "^1.0.0",
        "@itwin/tree-widget-react": "^1.0.0",
        "@itwin/web-viewer-react": "^4.0.0",
        "@itwin/webgl-compatibility": "^4.0.0",
        "@microsoft/applicationinsights-web": "^3.1.0",
        "@remix-run/router": "^1.3.2",
        "@types/applicationinsights": "^0.20.0",
        "@types/history": "^4.7.6",
        "@types/node": "^18.12.0",
        "@types/react": "^17.0.9",
        "@types/react-dom": "^17.0.9",
        "history": "^5.3.0",
        "react": "^18.2.0",
        "react-dom": "^18.2.0",
        "react-markdown": "^9.0.1",
        "react-redux": "^7.2.0",`

**Comment by mayank99** [comment]
This is unrelated to `react-markdown`.

`styles.css` and `useToaster` are only available in v3, whereas it looks like you are still using v2, judging by your snippet

> ```
> "@itwin/itwinui-react": "^2.7.0"
> ```

If you are ready to upgrade, you can check out our migration guide: https://github.com/iTwin/iTwinUI/wiki/iTwinUI-react-v3-migration-guide

**Comment by candcconsulting** [comment]
Thanks @mayank99 , that is strange as I had upgraded ... this is from my package.json
    "@itwin/imodels-access-frontend": "^4.0.0",
    "@itwin/itwinui-icons-color-react": "^2.1.0",
    "@itwin/itwinui-react": "^3.5.0",
    "@itwin/measure-tools-react": "^0.13.0",
    "@itwin/presentation-common": "^4.0.0",

**Reply by mayank99** [reply]
You can run `npm ls @itwin/itwinui-react` to find any accidental old versions that might be conflicting with the latest versions. You might need force overrides to work around it, which must be done carefully as it might break something else. See our guide on [Version conflicts](https://github.com/iTwin/iTwinUI/wiki/Version-conflicts) (it was written for v2, but the same general principles apply to v3 as well).

Again, the `react-markdown` part is unrelated, and likely just a coincidence.

**Reply by candcconsulting** [reply]
Here you go
PS F:\development\> npm ls @itwin/itwinui-react 
npm ERR! code ELSPROBLEMS
npm ERR! invalid: @itwin/itwinui-react@3.5.0 F:\development\node_modules\@itwin\itwinui-react
imodel-graph@0.1.1 F:\development\
├─┬ @itwin/appui-react@4.10.0
│ └── @itwin/itwinui-react@3.5.0 deduped
├─┬ @itwin/components-react@4.10.0
│ └── @itwin/itwinui-react@3.5.0 deduped
├─┬ @itwin/core-react@4.10.0
│ └── @itwin/itwinui-react@3.5.0 deduped
├─┬ @itwin/imodel-components-react@4.10.0
│ └── @itwin/itwinui-react@3.5.0 deduped
├── @itwin/itwinui-react@3.5.0
├─┬ @itwin/presentation-components@4.4.0
│ └── @itwin/itwinui-react@2.12.24
├─┬ @itwin/property-grid-react@1.5.1
│ └── @itwin/itwinui-react@3.5.0 deduped invalid: "^2.12.19" from node_modules/@itwin/property-grid-react
├─┬ @itwin/tree-widget-react@1.2.2
│ └── @itwin/itwinui-react@3.5.0 deduped invalid: "^2.12.19" from node_modules/@itwin/property-grid-react, "^2.11.0" from node_modules/@itwin/tree-widget-react
└─┬ @itwin/web-viewer-react@4.2.3
  └─┬ @itwin/viewer-react@4.6.2
    └── @itwin/itwinui-react@3.5.0 deduped invalid: "^2.12.19" from node_modules/@itwin/property-grid-react, "^2.11.0" from node_modules/@itwin/tree-widget-react, "^2.6.0" from node_modules/@itwin/viewer-react
    not sure I see a v2.7 in here

**Reply by mayank99** [reply]
I found 2.7.0 in your initial snippet. You may have changed something afterwards.

![image](https://github.com/iTwin/iTwinUI/assets/9084735/ffcd7408-4250-4d20-bebc-0626cca4145b)

In your latest snippet, I'm still seeing 2.12.24, which might be causing problems.

![image](https://github.com/iTwin/iTwinUI/assets/9084735/a75973fb-f85e-46ef-a0a2-d14898e61902)

It's hard for us to help with such insufficient details. You should attempt to debug this yourself and narrow down exactly what is causing the problem.

If possible, try creating a minimal reproduction (by forking [this sandbox](https://stackblitz.com/github/iTwin/iTwinUI/tree/main/minimal-sandbox?file=src/App.tsx)) to isolate the bug. In the process of isolating this issue, you might even end up fixing it yourself.

**Comment by candcconsulting** [comment]
ie my app runs ... If I install react-markdown I get the errors
